turtle_graphics/homework_2.js

```diff
@@ -0,0 +1,165 @@
+const fs = require("fs");
+
+class Turtle{
+    constructor(x,y){
+        this.x= x
+        this.y = y
+        this.result= [[this.x,this.y]]
+        this.direction='e'
+    }
+
+// change the value of x,y and push them to result array
+    forward(a){
+        if (this.direction=='n'){
+            for (let i=0; i<a; i++){
+                this.y= this.y-1
+                this.allPoint();
+            }
+        }else if(this.direction=='e'){
+            for (let i=0; i<a; i++){
+                this.x= this.x+1
+                this.allPoint();
+            }    
+        }else if(this.direction=='w'){
+            for (let i=0; i<a; i++){
+                this.x= this.x-1
+                this.allPoint();
+            }
+        }else if (this.direction=='s'){
+            for (let i=0; i<a; i++){
+                this.y= this.y+1
+                this.allPoint();
+            }
+        } else {
+            console.log('Something went wrong in forward function')
+        }
+        return this
+    }
+//set direction value n,e,w,s to control derection
+    right(){
+        if (this.direction=='n'){
+            this.direction = 'e'
+        }else if(this.direction=='e'){
+            this.direction = 's'
+        }else if(this.direction=='w'){
+            this.direction = 'n'
+        }else if (this.direction=='s'){
+            this.direction = 'w'
+        } else {
+            console.log('Something went wrong in rigth function')
+        }
+        return this 
+    }
+
+    left(){
+        if (this.direction=='n'){
+            this.direction = 'w'
+        }else if(this.direction=='e'){
+            this.direction = 'n'
+        }else if(this.direction=='w'){
+            this.direction = 's'
+        }else if (this.direction=='s'){
+            this.direction = 'e'
+        } else {
+            console.log('Something went wrong in left function')
+        }
+        return this 
+    }
+
+    allPoint(){
+        this.result.push([this.x,this.y])
+    }// record all the point simply by .push
```

Comment:
you can leave comments on complicated code on what it's doing but here the code itself should be self explanatory